subt_ign/src/GameLogicPlugin.cc

```diff
@@ -1578,18 +1578,18 @@ bool GameLogicPluginPrivate::OnNewArtifact(const subt::msgs::Artifact &_req,
     _resp.set_score_change(scoreDiff);
     _resp.set_report_status("scored");
 
-    if (!duplicate)
+    if (!duplicate) {
       this->totalScore += scoreDiff;
+      if (this->rosnode)
```

Comment:
@nkoenig can you confirm it's okay to move the publisher here so we only publish `artifactMsg` if the attempt was not a duplicate?

Reply:
This move is fine.

Reply:
That's fine.